sqlgen/start.go

```diff
@@ -193,22 +193,26 @@ func NewGenerator(state *State) func() string {
 						// all partitioned Columns should be contained in every unique/primary index.
 						c := partitionedCol.ToColumn()
 						Assert(c != nil)
+						if idx.Tp == IndexTypePrimary {
+							if c.defaultVal == "null" {
+								return Empty()
+							}
+							idx.RemoveColIf(func(col *Column) bool {
+								return col.defaultVal == "null"
+							})
+						}
```

Comment:
Is it possible that all the primary key columns are removed?

Can we solve this in the `GenNewIndex()`?

Reply:
I rewrite the code. PTAL again.